lib/spack/spack/database.py

```diff
@@ -691,10 +691,10 @@ def query(self, query_spec=any, known=any, installed=True, explicit=any, unique=
 
             if unique:
                 if len(results) > 1:
-                    tty.die('Found more than one spec for {}'.format(query_spec))
+                    tty.die('Found more than one spec for ' + str(query_spec))
```

Comment:
`'{}'.format()` only works for Python 2.7+. You have to use `'{0}'.format()` in Python 2.6+.